### PR TITLE
[plugin-kafka] Add ability to consume messages from Kafka

### PR DESCRIPTION
**Pull request by valfirst**

### CODE REVIEW

File: vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java [hunk 1]
```diff
@@ -58,4 +101,88 @@ public void sendData(String data, String topic) throws InterruptedException, Exe
...
+        BlockingQueue<String> messageQueue = new LinkedBlockingDeque<>();
+        testContext.put(MESSAGES_KEY, messageQueue);
+        ContainerProperties containerProperties = new ContainerProperties(topics.toArray(new String[0]));
+        containerProperties.setMessageListener(
+                (MessageListener<String, String>) data -> messageQueue.add(data.value()));
+        KafkaMessageListenerContainer<String, String> container = new KafkaMessageListenerContainer<>(consumerFactory,
+                containerProperties);
+        container.start();
+        testContext.put(LISTENER_KEY, container);
+        LOGGER.info("Kafka message listener is started");
```
Comment (ikalinin1) [inline vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java:123]: does it make sense to add topics into the log message?

File: vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java [hunk 2]
```diff
@@ -58,4 +101,88 @@ public void sendData(String data, String topic) throws InterruptedException, Exe
...
+     *                       (<i>Possible values:<b> less than, less than or equal to, greater than, greater than or
+     *                       equal to,
+     *                       equal to</b></i>)
+     * @param expectedCount  the expected count of the messages to be matched by the rule
+     */
+    @When("I wait with `$timeout` timeout until count of consumed Kafka messages is $comparisonRule `$expectedCount`")
+    public void waitForKafkaMessages(Duration timeout, ComparisonRule comparisonRule, int expectedCount)
+    {
+        Matcher<Integer> countMatcher = comparisonRule.getComparisonRule(expectedCount);
+        Integer result = new DurationBasedWaiter(timeout, Duration.ofSeconds(1)).wait(
```
Comment (ikalinin1) [inline vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java:141]: Could you please clarify why the polling duration is a hardcoded value?

File: vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java (hunk 1 above)
Comment (valfirst) [reply vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java:123]: the data is logged by kafka client:
log

File: vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java (hunk 2 above)
Comment (valfirst) [reply vividus-plugin-kafka/src/main/java/org/vividus/bdd/steps/kafka/KafkaSteps.java:141]: we poll the data from memory (`testContext`), so it's cheap, 
IMHO, 1 second is a period which could save test time and at the same time it's not so much for humans (like, 1s or 0.5s - it doesn't matter for users)

**Review by ikalinin1 — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

File: vividus-plugin-kafka/src/test/java/org/vividus/bdd/steps/kafka/KafkaStepsTests.java [hunk 3]
```diff
@@ -16,68 +16,85 @@
...
+    private final TestLogger logger = TestLoggerFactory.getTestLogger(KafkaSteps.class);
 
     @Mock private IPropertyParser propertyParser;
+    @Mock private TestContext testContext;
+    @Mock private IBddVariableContext bddVariableContext;
+    @Mock private SoftAssert softAssert;
+    private KafkaSteps kafkaSteps;
+
+    @BeforeEach
+    void beforeEach()
```
Comment (uarlouski) [inline vividus-plugin-kafka/src/test/java/org/vividus/bdd/steps/kafka/KafkaStepsTests.java:59]: https://github.com/vividus-framework/vividus/blob/master/CONTRIBUTING.md#unit-tests

**Pull request merged**